File: Data Structures & Algorithms/climbing-stairs/submission-0.py
class Solution:
    def climbStairs(self, n: int) -> int:
        res = 0
        for i in range (n+1):
            a = i
            b =  (n - a) % 2

            if(a == 0 or a == n):
                if(b == 0):
                    res += 1
            else:
                if(b == 0):
                    res += self.count_arrangements(a, int((n-a)/2))

        return res

    def count_arrangements(self, a, b):
        return math.comb(a + b, a)

File: Data Structures & Algorithms/max-water-container/submission-0.java
class Solution {
    public int maxArea(int[] heights) {
        int l = 0;
        int r = heights.length - 1;
        int res = 0;
        while(l <= r) {
            int temp = Math.min(heights[l], heights[r]);
            res = Math.max(res, (r - l) * temp);
            if(temp == heights[l]) l++;
            else r--;
        }
        return res;
    }
}
